src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -2813,6 +2875,9 @@ public async Task HubMethodInvokeDoesNotCountTowardsClientTimeout()
 
                 using (var client = new TestClient(new JsonHubProtocol()))
                 {
+                    var customDuplex = new CustomDuplex(client.Connection.Transport);
+                    client.Connection.Transport = customDuplex;
+
                     var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
                     // This starts the timeout logic
                     await client.SendHubMessageAsync(PingMessage.Instance);
```

Comment:
Is there a risk that the client will timeout between sending the ping and invoking the long running method given the ClientTimeoutInterval is 0 seconds? Can we use a mock system clock or something like that?

Reply:
> Is there a risk that the client will timeout between sending the ping and invoking the long running method given the ClientTimeoutInterval is 0 seconds? Can we use a mock system clock or something like that?

No risk because that logic lives in `HttpConnectionManager` and we are only working with the `HubConnectionHandler` layer.